### ISSUE TITLE
MF6 LAW=1: multi-panel jax.grad wrt incident energy E (avoid lax.switch compile stall)

### ISSUE THREAD

**Issue by gschnabel** [IAEA-NDS/endf-userpy#166]
## Context

PR #164 added ``panel_idx=`` autodiff entry points for the LAW=1 spectrum and DDX paths, enabling ``jax.grad`` wrt incident energy ``E``, outgoing energy ``E'``, and ``mu`` when the query stays inside a single panel-pair. Callers who need grad across panel knots must currently:

- compute ``panel_idx = find_interval(ei_mesh, [E_concrete])[0]`` themselves from a concrete ``E``, then
- call the function with that ``panel_idx=`` and ``jax.grad`` inside that panel.

This works but requires knowing the panel before differentiating. It cannot be composed with an outer optimizer that lets ``E`` roam freely across panels.

## What we prototyped and why it did not ship

A JAX multi-panel auto-dispatch was implemented and verified numerically correct (numpy vs jax matched to ``5e-16`` relative on Al-27 MT=91), then dropped from PR #164 because of XLA compile time. Sketch of the approach:

```python
def _integrate_law1_spectrum_multipanel_jax(data, energies_in, energies_out, ...):
    n_pairs = int(data.ei_mesh.shape[0]) - 1
    ei_mesh_j = xp.asarray(data.ei_mesh)

    def make_branch(p):
        def branch(e_scalar):
            return _law1_spectrum_panel_pair(
                data, p, lei_list[p], e_scalar[None], ep_out_xp,
                eff_lct, gl_x, gl_w, xp,
            )[0]
        return branch
    branches = [make_branch(p) for p in range(n_pairs)]

    def per_e(e_scalar):
        idx = xp.clip(
            xp.searchsorted(ei_mesh_j, e_scalar, side='right') - 1,
            0, n_pairs - 1,
        )
        return jax.lax.switch(idx, branches, e_scalar)

    return xp.stack([per_e(e_arr[i]) for i in range(n_e)], axis=0)
```

**The problem.** ``lax.switch`` compiles every branch separately, and every branch has a different intermediate-array shape (the kink count ``nep1 + nep2`` varies per panel-pair, so ``ep_panels[p, :nep1]``, kink arrays, subpanel arrays and GL-node arrays all take on different sizes in different branches). For a realistic section with ~16 panel-pairs the XLA compile step stalls past 2 minutes on first call, which is unusable for a smoke test and would break CI. Repeated calls in the same Python process do hit the compilation cache after that, but the first-call cost makes it a bad default.

## Path forward: single-graph tracer-indexed kernel

Refactor ``_law1_spectrum_panel_pair`` to accept the panel index as a tracer int and use xp-native indexing into pre-stacked arrays instead of Python-side integer indexing on ``data``:

- ``ei_mesh``, ``ep_panels``, ``b_panels``, ``nep_arr``, ``nd_arr``, ``na_arr`` all converted to xp-native at preproc time (only ``b_panels`` is today; the others are numpy).
- Replace ``e1 = float(data.ei_mesh[p1])`` with a jax gather ``ei_mesh_j[p1]``, and similarly for the panel-level shape parameters.
- Replace ``ep1_arr = xp.asarray(data.ep_panels[p1, :nep1])`` (dynamic slicing on tracer ``nep1``) with the full padded row ``ep_panels[p1]`` shape ``(max_nep,)``. Mask padded entries so their kink contributions clamp to ``umin`` (zero-width subpanels), matching the pattern already used for ``b_panels``.

The result is a single XLA-compiled kernel parameterized by a tracer ``panel_idx``. Compile cost then scales like one kernel instead of ``n_pairs`` kernels. ``jnp.searchsorted`` chooses the panel per ``E``, no ``lax.switch`` needed.

Some care needed for:

- The ``if nep1 <= nd1 or nep2 <= nd2: return zeros`` early-out becomes an ``xp.where``; not a real branch under tracing.
- The ``last valid Ep of panel`` gather (``ep_panels[p1, nep1-1]``) uses ``jnp.take`` with a tracer index.
- Padded-row masking must preserve the sort behaviour so zero-width subpanels do not add spurious contributions.

## Scope

- **In:** spectrum integrator (``integrate_law1_spectrum``) and DDX reconstruction (``mf6_law1_kernel.reconstruct``). Both currently only expose single-panel autodiff via ``panel_idx=``.
- **Out (initial):** LAW=6/7, MF4/MF5. Extend once the pattern is validated on LAW=1.
- **Out (initial):** grad across panel knots is physically discontinuous (real C0 kinks at each ``ei_mesh`` knot). This work does not change that; it enables one-sided grad on either side of a knot to be computed automatically instead of by the caller.

## Deliverables

1. Refactor ``_law1_spectrum_panel_pair`` for tracer ``panel_idx``.
2. Auto-dispatch in ``integrate_law1_spectrum`` (and ``reconstruct``) when ``xp`` is JAX and no ``panel_idx`` is provided.
3. Tests: multi-panel ``jax.grad`` wrt ``E`` at points on both sides of a ``ei_mesh`` knot, matching central finite-difference to the interior tol we already use elsewhere.
4. Benchmark first-call compile time and steady-state runtime vs the current single-panel entry.

## Priority

Low. Callers who need multi-panel autodiff can dispatch ``panel_idx`` themselves today (one line per call). Only worth doing if a downstream user explicitly asks for it, or if we start building JAX-native optimizers over the LAW=1 continuum that need to move ``E`` freely across panels.

## Related

- Landed: PR #161 (kernel), #162 (backend-agnostic), #163 (kink-aware GL), #164 (single-panel autodiff), #165 (adaptive linearization).
- Umbrella: issue #154 (JAX autodiff for MF6 LAW=1).
